### PR TITLE
feat: browserless login

### LINKED ISSUE #60: Add a way to login with username and password

### PR DESCRIPTION
**Pull request by volgar1x**
In some cases, a user might not have access to a browser but still need to login through `aqora`.
This PR adds a fallback for those case, using `passterm@2.0` in order to prompt passwords securely.

### CODE REVIEW

**Review by jpopesculian — CHANGES_REQUESTED** [review]
This looks great! Given that `open` can be kind of unpredictable, I would just add a flag to the login command to force the use of interactive login if so desired

**Comment by volgar1x** [comment]
Fixes https://github.com/aqora-io/cli/issues/60 btw

File: src/commands/login.rs [hunk 1]
```diff
@@ -98,54 +102,218 @@ where
...
-        if open::that(authorize_url.as_str()).is_err() {
-            cloned_progress.set_message(format!(
-                "Failed to open browser, please open {authorize_url} manually"
-            ));
-        }
-        cloned_progress.set_message("Waiting for browser response...");
-    });
+
+    if !interactive {
+        progress.suspend(|| {
```
Comment (jpopesculian) [inline src/commands/login.rs:140]: you can also just use [progress.println](https://docs.rs/indicatif/latest/indicatif/struct.ProgressBar.html#method.println)

File: src/commands/login.rs [hunk 2]
```diff
@@ -28,7 +29,10 @@ const CLIENT_ID_PREFIX: &str = "localhost-";
 
 #[derive(Args, Debug, Serialize)]
 #[command(author, version, about)]
-pub struct Login;
+pub struct Login {
+    #[arg(long, short)]
+    interactive: bool,
```
Comment (jpopesculian) [inline src/commands/login.rs:34]: something like "no-browser" would maybe be more clear. or at least put a little help message to explain

File: src/commands/login.rs [hunk 3]
```diff
@@ -98,54 +102,218 @@ where
...
-            ));
-        }
-        cloned_progress.set_message("Waiting for browser response...");
-    });
+
+    if !interactive {
+        progress.suspend(|| {
+            // NOTE: suspending here here instead of just `progress.println(...)`
+            // because indicatif will cut extra characters instead of wrapping
+            println!("Please navigate to this url (it should be done automatically):");
```
Comment (jpopesculian) [inline src/commands/login.rs:143]: I would just say `Opening {authorize_url} (it should open automatically)` that way people don't thing they have to do anything. Maybe also add a helpful message for people that don't have access to a browser to use `--interactive` or whatever

**Review by jpopesculian — CHANGES_REQUESTED** [review]
Looks awesome! Just a couple of clarifying things

File: src/commands/login.rs (hunk 1 above)
Comment (volgar1x) [reply src/commands/login.rs:140]: I added a note explaining why I did not use [progress.println](https://docs.rs/indicatif/latest/indicatif/struct.ProgressBar.html#method.println)

File: src/commands/login.rs (hunk 3 above)
Comment (volgar1x) [reply src/commands/login.rs]: > I would just say Opening {authorize_url} (it should open automatically) that way people don't thing they have to do anything.

`authorize_url` is a very long string, so I decided to put on its line in order to clearly show `(it should open automatically)`

> Maybe also add a helpful message for people that don't have access to a browser to use --interactive or whatever

Great idea 🙂 added it!

File: src/commands/login.rs (hunk 2 above)
Comment (volgar1x) [reply src/commands/login.rs:34]: I added a help message

File: src/commands/login.rs (hunk 1 above)
Comment (jpopesculian) [reply src/commands/login.rs:140]: oh sorry, my bad, i didn't read your note correctly! makes sense

**Review by jpopesculian — APPROVED** [review]

**Pull request merged**